### PR TITLE
Default preferences for channels in template

### PR DESCRIPTION
**Pull request by ainouzgali**

### CODE REVIEW

File: apps/api/src/app/widgets/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts [hunk 1]
```diff
@@ -61,10 +64,9 @@ export class GetSubscriberTemplatePreference {
   }
 }
 
-function filterActiveChannels(channels: IPreferenceChannels, activeChannels: ChannelTypeEnum[]): IPreferenceChannels {
-  const filteredChannels = Object.assign({}, channels);
-
-  for (const key in channels) {
+function getDefaultFromTemplate(activeChannels: ChannelTypeEnum[], defaults: IPreferenceChannels) {
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts:67]: Maybe worth returning here a particular type: `IPreferenceChannels` or something like this

File: libs/dal/src/repositories/notification-template/notification-template.schema.ts [hunk 2]
```diff
@@ -97,6 +97,28 @@ const notificationTemplateSchema = new Schema(
...
+        type: Schema.Types.Boolean,
+        default: true,
+      },
+      direct: {
+        type: Schema.Types.Boolean,
+        default: true,
+      },
+      push: {
+        type: Schema.Types.Boolean,
+        default: true,
```
Comment (scopsy) [inline libs/dal/src/repositories/notification-template/notification-template.schema.ts:119]: Good to also have the defaults here, love it!

**Review by scopsy — COMMENTED** [review]
Looks awesome @ainouzgali ! Added a couple of comments/questions 🚀

File: apps/api/src/app/widgets/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts [hunk 3]
```diff
@@ -37,7 +37,10 @@ export class GetSubscriberTemplatePreference {
         template: responseTemplate,
         preference: {
           enabled: currSubscriberPreference.enabled,
-          channels: filterActiveChannels(currSubscriberPreference.channels, activeChannels),
+          channels: filterActiveChannels(
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts:40]: Maybe worth adding tests for this in the existing tests too?

Comment (ainouzgali) [reply apps/api/src/app/widgets/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts:40]: Yes, almost ready. Will push soon.

**Pull request merged**